### PR TITLE
Allow skipping builds for some packages on ARM64

### PR DESCRIPTION
**Pull request by andrewlock**
## Summary of changes
- Make skipping builds on Alpine more granular (allow skipping for certain packages only)
- Allow skipping builds on ARM64

## Reason for change

The `Grpc.Tools` package doesn't work on ARM64 until package version 2.37.0. Also, it doesn't _currently_ support Alpine, so it may in the future. This PR is a blocker for #2535 and #2545

## Implementation details

Moved the `SkipAlpine` flag into the package version conditions, and added a `SkipArm64` flag.

## Test coverage

Tested in [the Grpc branch here](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93388&view=results)

### CODE REVIEW

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2562 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`WriteAndFlushEnrichedTraces`|net472|863μs|2.71μs|9.38μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|677μs|3.73μs|21.4μs|0| 0|0|2.57&nbsp;KB|
|#2562|`WriteAndFlushEnrichedTraces`|net472|871μs|2.95μs|12.9μs|0| 0|0|3.16&nbsp;KB|
|#2562|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|688μs|3.88μs|28.2μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`SendRequest`|netcoreapp3.1|326μs|522ns|1.95μs|0.163| 0|0|19.74&nbsp;KB|
|#2562|`SendRequest`|net472|0.0216ns|0.00781ns|0.0615ns|0| 0|0|0&nbsp;b|
|#2562|`SendRequest`|netcoreapp3.1|332μs|842ns|3.04μs|0.167| 0|0|19.74&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`ExecuteNonQuery`|net472|1.82μs|8.04ns|30.1ns|0.0939| 0.000886|0|594&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`ExecuteNonQuery`|netcoreapp3.1|1.5μs|6.49ns|25.1ns|0.00888| 0|0|632&nbsp;B|
|#2562|`ExecuteNonQuery`|net472|1.77μs|9.02ns|39.3ns|0.0936| 0.000859|0|594&nbsp;B|
|#2562|`ExecuteNonQuery`|netcoreapp3.1|1.48μs|4.05ns|14.6ns|0.00882| 0|0|632&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`CallElasticsearch`|net472|2.73μs|13.3ns|58.1ns|0.127| 0|0|803&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`CallElasticsearch`|netcoreapp3.1|1.76μs|8.36ns|33.5ns|0.0104| 0|0|792&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`CallElasticsearchAsync`|net472|3.01μs|11.8ns|51.3ns|0.148| 0|0|939&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`CallElasticsearchAsync`|netcoreapp3.1|1.79μs|6.4ns|24.8ns|0.0124| 0|0|912&nbsp;B|
|#2562|`CallElasticsearch`|net472|2.83μs|12.1ns|45.3ns|0.126| 0|0|802&nbsp;B|
|#2562|`CallElasticsearch`|netcoreapp3.1|1.74μs|7.01ns|27.2ns|0.0106| 0|0|792&nbsp;B|
|#2562|`CallElasticsearchAsync`|net472|2.96μs|9.37ns|33.8ns|0.149| 0.00147|0|939&nbsp;B|
|#2562|`CallElasticsearchAsync`|netcoreapp3.1|1.89μs|9.12ns|40.8ns|0.0129| 0|0|912&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`ExecuteAsync`|net472|3.41μs|18.2ns|98.1ns|0.168| 0.0016|0|1.06&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`ExecuteAsync`|netcoreapp3.1|2μs|8.47ns|30.5ns|0.0137| 0|0|1.03&nbsp;KB|
|#2562|`ExecuteAsync`|net472|3.31μs|11.4ns|42.5ns|0.168| 0.00164|0|1.06&nbsp;KB|
|#2562|`ExecuteAsync`|netcoreapp3.1|1.99μs|5.65ns|20.4ns|0.0139| 0|0|1.03&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`SendAsync`|net472|7.16μs|36ns|165ns|0.36| 0|0|2.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`SendAsync`|netcoreapp3.1|4.93μs|21.4ns|93.1ns|0.0293| 0|0|2.21&nbsp;KB|
|#2562|`SendAsync`|net472|7.17μs|32.1ns|124ns|0.359| 0|0|2.28&nbsp;KB|
|#2562|`SendAsync`|netcoreapp3.1|4.74μs|22.7ns|90.7ns|0.0309| 0|0|2.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|net472|3.75μs|20.9ns|129ns|0.227| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|netcoreapp3.1|3.19μs|13ns|50.3ns|0.0205| 0|0|1.53&nbsp;KB|
|#2562|`EnrichedLog`|net472|3.74μs|22.7ns|222ns|0.227| 0|0|1.45&nbsp;KB|
|#2562|`EnrichedLog`|netcoreapp3.1|3.21μs|15ns|57.9ns|0.0205| 0|0|1.53&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|net472|300μs|1.37μs|5.49μs|0.444| 0.148|0|4.33&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|netcoreapp3.1|236μs|699ns|2.71μs|0| 0|0|4.21&nbsp;KB|
|#2562|`EnrichedLog`|net472|306μs|1.44μs|5.93μs|0.452| 0.151|0|4.33&nbsp;KB|
|#2562|`EnrichedLog`|netcoreapp3.1|240μs|539ns|2.02μs|0| 0|0|4.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|net472|7.9μs|38.3ns|158ns|0.504| 0|0|3.23&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|netcoreapp3.1|6.71μs|26ns|97.2ns|0.0497| 0|0|3.6&nbsp;KB|
|#2562|`EnrichedLog`|net472|8.32μs|43.3ns|212ns|0.505| 0|0|3.23&nbsp;KB|
|#2562|`EnrichedLog`|netcoreapp3.1|7μs|25.1ns|94ns|0.049| 0|0|3.6&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`SendReceive`|net472|2.39μs|9.49ns|36.7ns|0.16| 0|0|1.01&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`SendReceive`|netcoreapp3.1|2.17μs|12.6ns|110ns|0.0139| 0|0|1.01&nbsp;KB|
|#2562|`SendReceive`|net472|2.5μs|9.61ns|35.9ns|0.16| 0.00123|0|1.01&nbsp;KB|
|#2562|`SendReceive`|netcoreapp3.1|2.08μs|6.96ns|27ns|0.0142| 0|0|1.01&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|net472|6.4μs|31.1ns|132ns|0.289| 0|0|1.87&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`EnrichedLog`|netcoreapp3.1|5.78μs|37ns|367ns|0.0187| 0|0|1.49&nbsp;KB|
|#2562|`EnrichedLog`|net472|6.65μs|26ns|93.7ns|0.291| 0|0|1.87&nbsp;KB|
|#2562|`EnrichedLog`|netcoreapp3.1|5.59μs|25.3ns|97.8ns|0.0215| 0|0|1.49&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`StartFinishSpan`|net472|971ns|3.26ns|12.2ns|0.0716| 0|0|457&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`StartFinishSpan`|netcoreapp3.1|902ns|4.26ns|16.5ns|0.00625| 0|0|456&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`StartFinishScope`|net472|1.14μs|4.62ns|17.9ns|0.0841| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/8b2fe252ce569a3a33fe5eb522e392aa5078f9a1)|`StartFinishScope`|netcoreapp3.1|1.07μs|5.43ns|24.3ns|0.00795| 0|0|576&nbsp;B|
|#2562|`StartFinishSpan`|net472|970ns|5.39ns|35ns|0.0716| 0|0|457&nbsp;B|
|#2562|`StartFinishSpan`|netcoreapp3.1|921ns|5.01ns|29.6ns|0.00656| 0|0|456&nbsp;B|
|#2562|`StartFinishScope`|net472|1.24μs|6.82ns|39.2ns|0.0843| 0|0|538&nbsp;B|
|#2562|`StartFinishScope`|netcoreapp3.1|1.06μs|4.36ns|16.3ns|0.00798| 0|0|576&nbsp;B|
</details>

**Review by pierotibou — APPROVED** [review]

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2562 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/c01221e04ec68bcb73c8e6b42be7b0e29c53af5f) will **not change** line coverage
:heavy_check_mark: Merging #2562 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/c01221e04ec68bcb73c8e6b42be7b0e29c53af5f) will **not change** branch coverage
:heavy_check_mark: Merging #2562 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/c01221e04ec68bcb73c8e6b42be7b0e29c53af5f) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/c01221e04ec68bcb73c8e6b42be7b0e29c53af5f) | #2562       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `12310` / `17219` | `12292` / `17219` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `6746` / `10129` | `6739` / `10129` |          |
| Branches %| `67%`      | `67%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `11340`      | `11340`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93310&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93310/artifacts?artifactName=Code%20Coverage%20Report_93310&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2562](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93406&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93406/artifacts?artifactName=Code%20Coverage%20Report_93406&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/c01221e04ec68bcb73c8e6b42be7b0e29c53af5f) | #2562       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `67%`      | `67%`       |  `0%` :heavy_check_mark:  |
| Complexity| `11340`      | `11340`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Telemetry.JsonHttpClientTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/b61837146f60bc15e78007b63e66b2005880fffb/tracer/src/Datadog.Trace/Telemetry/Transports/JsonHttpClientTelemetryTransport.cs) | `-26%` :no_entry: | `-38%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Telemetry.JsonWebRequestTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/b61837146f60bc15e78007b63e66b2005880fffb/tracer/src/Datadog.Trace/Telemetry/Transports/JsonWebRequestTelemetryTransport.cs) | `28%` :heavy_check_mark: | `57%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93310&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93310/artifacts?artifactName=Code%20Coverage%20Report_93310&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2562](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93406&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93406/artifacts?artifactName=Code%20Coverage%20Report_93406&api-version=6.0&%24format=zip)

File: tracer/build/PackageVersionEntrySchema.json [hunk 1]
```diff
@@ -64,15 +64,19 @@
                       "uniqueItems": true
                     }
                   ]
+                },
+                "SkipAlpine": {
+                  "type": "boolean",
+                  "description": "If true, packages in the range will not built on Alpine"
```
Comment (anna-git) [inline tracer/build/PackageVersionEntrySchema.json:70]: nit: will not be built on Alpine

File: tracer/build/_build/GeneratePackageVersions/PackageVersionEntry.cs [hunk 2]
```diff
@@ -55,6 +49,16 @@ public record PackageVersionConditionEntry
...
             public TargetFramework[] ExcludeTargetFrameworks { get; init; } = Array.Empty<TargetFramework>();
             public TargetFramework[] IncludeOnlyTargetFrameworks { get; init; } = Array.Empty<TargetFramework>();
+
+            /// <summary>
+            /// If true, packages in the range will not be built on Arm64
+            /// </summary>
+            public bool SkipArm64 { get; init; }
+
+            /// <summary>
+            /// If true, packages in the range will not built on Alpine
```
Comment (anna-git) [inline tracer/build/_build/GeneratePackageVersions/PackageVersionEntry.cs:59]: nit: will not be built on Alpine

**Review by anna-git — APPROVED** [review]

File: tracer/build/PackageVersionEntrySchema.json (hunk 1 above)
Comment (andrewlock) [reply tracer/build/PackageVersionEntrySchema.json:70]: Thanks will fix in a separate PR to avoid another CI cycle 🙂

**Pull request merged**